authentication/auth.go

```diff
@@ -527,16 +509,32 @@ func GenerateTokenFromDefaultsV2(schemaURL string, accountServiceURL string, url
 	if claims["survey_metadata"] != nil {
 		surveyMetadata = claims["survey_metadata"].(map[string]interface{})
 	}
-	for key, value := range data {
-		updatedData[key] = value
-	}
 
 	initialData := surveyMetadata["data"].(map[string]interface{})
 
 	for key, value := range initialData {
 		updatedData[key] = value
 	}
 
+	/*
+		This method below is used to add any metadata (that is a parameter) which is intended to be of type boolean to
+		requiredSchemaMetadata for the typing in "updatedData" to be correct. Without this method, a parameter added to
```

Comment:
Maybe I'm misunderstanding but `requiredSchemaMetadata` should only refer to metadata that is specified in the actual schema, I don't think we need to add any values to it?